docs/csharp/language-reference/operators/bitwise-and-shift-operators.md

```diff
@@ -0,0 +1,182 @@
+
+You can also use the `~` symbol to declare finalizers. For more information, see [Finalizers](../../programming-guide/classes-and-structs/destructors.md).
+
+## Left-shift operator \<\<
+
+The `<<` operator shifts its first operand left by the number of bits defined by its second operand.
+
+The left-shift operation discards the high-order bits that are outside the range of the result type and sets the low-order empty bit positions to zero, as the following example shows:
+
+[!code-csharp-interactive[left shift](~/samples/snippets/csharp/language-reference/operators/BitwiseAndShiftOperators.cs#LeftShift)]
+
+For information about how the second operand of the `<<` operator defines the shift count, see the [Shift count of the shift operators](#shift-count-of-the-shift-operators) section.
+
+## Right-shift operator >>
+
+The `>>` operator shifts its first operand right by the number of bits defined by its second operand.
+
+The right-shift operation discards the low-order bits, as the following example shows:
+
+[!code-csharp-interactive[right shift](~/samples/snippets/csharp/language-reference/operators/BitwiseAndShiftOperators.cs#RightShift)]
+
+The high-order empty bit positions are set based on the type of the first operand as follows:
+
+- If the first operand is of type [int](../keywords/int.md) or [long](../keywords/long.md), the right-shift operator performs an *arithmetic* shift: the value of the most significant bit (the sign bit) of the first operand is propagated to the high-order empty bit positions. That is, the high-order empty bit positions are set to zero if the first operand is non-negative and set to one if it's negative.
+
+  [!code-csharp-interactive[arithmetic right shift](~/samples/snippets/csharp/language-reference/operators/BitwiseAndShiftOperators.cs#ArithmeticRightShift)]
+
+- If the first operand is of type [uint](../keywords/uint.md) or [ulong](../keywords/ulong.md), the right-shift operator performs a *logical* shift: the high-order empty bit positions are always set to zero.
+
+  [!code-csharp-interactive[logical right shift](~/samples/snippets/csharp/language-reference/operators/BitwiseAndShiftOperators.cs#LogicalRightShift)]
+
+For information about how the second operand of the `>>` operator defines the shift count, see the [Shift count of the shift operators](#shift-count-of-the-shift-operators) section.
+
+## Logical AND operator &amp;
+
+The `&` operator computes the bitwise logical AND of its operands:
+
+[!code-csharp-interactive[bitwise AND](~/samples/snippets/csharp/language-reference/operators/BitwiseAndShiftOperators.cs#BitwiseAnd)]
+
+For the operands of the `bool` type, the `&` operator computes the [logical AND](boolean-logical-operators.md#logical-and-operator-) of its operands. The unary `&` operator is the [address-of operator](and-operator.md#unary-address-of-operator).
```

Comment:
Alternative phrasing to start the sentence: "For the `bool` operands,..." Should we switch to this one? If yes, then how to talk about other types:
- "For the numeric (integral) type operands..."
- "For the numeric (integral) operands..." (sounds best to me)
- "For numeric (integral) operands..."
- Or just use the existing approach: "For the operands of the numeric (integral) types..."

Reply:
I like the current phrasing: This article is about the bitwise operators, so pointing out that `bool` is different, with a link, feels natural.

Also, it avoids the fact that it's not all *integral types*. This article applies to `int`, `uint`, `long`, and `ulong`.